mirror_builder/__main__.py

```diff
@@ -51,5 +75,41 @@ def do_bootstrap(ctx, args):
         sdist.handle_requirement(ctx, Requirement(toplevel))
 
 
+def do_download_source_archive(ctx, args):
+    req = Requirement(f'{args.dist_name}=={args.dist_version}')
+    filename, _ = sources.download_source(ctx, req)
+    with open(ctx.work_dir / 'last-download.txt', 'w') as f:
+        f.write(filename)
+    print(filename)
+
+
+def do_prepare_source(ctx, args):
+    req = Requirement(f'{args.dist_name}=={args.dist_version}')
+    source_filename = pathlib.Path(args.source_archive)
+    # FIXME: Does the version need to be a Version instead of str?
+    source_root_dir = sources.prepare_source(ctx, req, source_filename, args.dist_version)
+    with open(ctx.work_dir / 'last-source-dir.txt', 'w') as f:
+        f.write(str(source_root_dir))
+    print(source_root_dir)
+
+
+def do_prepare_build(ctx, args):
+    server.start_wheel_server(ctx)
+    req = Requirement(f'{args.dist_name}=={args.dist_version}')
+    source_root_dir = pathlib.Path(args.source_dir)
+    sdist.prepare_build_environment(ctx, req, source_root_dir)
+
+
+def do_build(ctx, args):
+    req = Requirement(f'{args.dist_name}=={args.dist_version}')
+    source_root_dir = pathlib.Path(args.source_dir)
+    build_env = wheels.BuildEnvironment(ctx, source_root_dir.parent, None)
+    wheel_filenames = wheels.build_wheel(ctx, req, source_root_dir, build_env)
```

Comment:
I know it's pre-existing, but this thing about "user: build me a wheel", "program: here's a list of wheels that we expect to only have one element" is pretty annoying

Reply:
Filed as #66